src/lib/blueprint.ts

```diff
@@ -307,7 +311,11 @@ const createRoutes = async (
     }
   }
 
-  return Array.from(routeMap.values())
+  let routes = Array.from(routeMap.values())
+
+  routes = routes.map(addIsDeprecatedToRoute).map(addIsUndocumentedToRoute)
+
+  return routes.map(addNamespaceStatusToRoute(routes))
```

Comment:
can't chain these as `addNamespaceStatusToRoute` needs updated routes after `addIsDeprecatedToRoute` and `addIsUndocumentedToRoute`

Reply:
JS map actually passes the array to the callback: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map#array

So you can do this without a higher order function / let binding with

```
routes.map(addIsDeprecatedToRoute).map(addIsUndocumentedToRoute).map(addNamespaceStatusToRoute)
```

if you define

```
const addNamespaceStatusToRoute = (routes: Route, _idx: number, routes: Route[]): Route => {

Reply:
Cool! Thanks for sharing this!

Reply:
e211441f21c7728b1857207f2747b33c03fc4a80